### PR TITLE
Add missing parameter types

### PR DESCRIPTION
**Pull request by danielrearden**

### CODE REVIEW

File: src/renderGraphiQL.js [hunk 1]
```diff
@@ -19,9 +19,9 @@ export type GraphiQLOptions = {|
 |};
 
 // Ensures string values are safe to be used within a <script> tag.
-function safeSerialize(data) {
+function safeSerialize(data: mixed) {
```
Comment (IvanGoncharov) [inline src/renderGraphiQL.js:22]:
```suggestion
function safeSerialize(data: mixed): string {
```
Also, return values on those functions don't cause any issue with Flow we can add them.

File: src/parseBody.js [hunk 2]
```diff
@@ -76,7 +76,10 @@ export async function parseBody(
 const jsonObjRegex = /^[ \t\n\r]*\{/;
 
 // Read and parse a request body.
-async function readBody(req, typeInfo) {
+async function readBody(
+  req: $Request,
+  typeInfo: {| type: string, parameters: { [param: string]: string, ... } |},
```
[the remark this answers is not in the document]
Comment (IvanGoncharov) [reply src/parseBody.js:81]: It's actually helpful to have `TODO` in code for such a thing.

File: src/__tests__/http-test.js [hunk 3]
```diff
@@ -75,11 +76,13 @@ function urlString(urlParams?: { [param: string]: string, ... }): string {
   return string;
 }
 
-[
+const implementations: Array<[(...args: Array<any>) => any, string]> = [
```
Comment (IvanGoncharov) [inline src/__tests__/http-test.js:79]: This code was a constant source of frustration for me and now it even introduces bunch of any types.
So I just rewrote it in #631 
Can you please rebase?

File: src/__tests__/http-test.js [hunk 4]
```diff
@@ -1962,7 +1963,9 @@ function runTests(server) {
   });
 
   describe('Custom validation rules', () => {
-    const AlwaysInvalidRule = function (context) {
+    const AlwaysInvalidRule = function (
+      context: ValidationContext,
+    ): { [key: string]: mixed, ... } {
```
Comment (IvanGoncharov) [inline src/__tests__/http-test.js:1968]: You can use `ASTVisitor` as return value.

File: src/renderGraphiQL.js [hunk 5]
```diff
@@ -19,9 +19,9 @@ export type GraphiQLOptions = {|
 |};
 
 // Ensures string values are safe to be used within a <script> tag.
-function safeSerialize(data) {
+function safeSerialize(data: mixed): string {
```
Comment (IvanGoncharov) [inline src/renderGraphiQL.js:22]: Actually Flow is correct and you don't need `any` if you change the argument to `data: string | null`

**Comment by IvanGoncharov** [comment]
@danielrearden Left a couple of comments and everything else looks great 👍

File: src/renderGraphiQL.js (hunk 5 above)
Comment (danielrearden) [reply src/renderGraphiQL.js]: Oops, I thought we were passing other types here, not just strings. I ended up using `?string` here to match the types of the fields in `GraphiQLData`.

**Pull request merged**